code/modules/hydroponics/hydroponics_mutations.dm

```diff
@@ -0,0 +1,529 @@
+//The astrogator has a way with plants of every type.
+//She nurtures alien veg'tables and cooks them when they're ripe!
+//She had a Veged orchid once that ate the captain's cat.
+//Then terrorized the ship's exec, until she squashed it flat!
+
+//Mutates the plant overall (randomly).
+/obj/machinery/portable_atmospherics/hydroponics/proc/mutate(var/severity)
+
+	//writepanic("[__FILE__].[__LINE__] ([src.type])([usr ? usr.ckey : ""])  \\/obj/machinery/portable_atmospherics/hydroponics/proc/mutate() called tick#: [world.time]")
+
+	if(!severity) return
+
+	// No seed? Try to mutate the weeds or pests in the tray, if any.
+	if(!seed)
+		//TODO
+		return
+
+	if(seed.immutable)
+		return
+
+	//Is the plant still a sapling? If so, try to mutate species, otherwise do something bad.
+	if(age < 3)
+		if(seed.mutants. && seed.mutants.len)
+			if(prob(30))
+				mutate_species()
+				return
+		var/mutation_type = pick_mut(severity, "SomethingBad")
+		apply_mut(mutation_type, severity)
+		return
+
+	//writepanic("[__FILE__].[__LINE__] ([src.type])([usr ? usr.ckey : ""])  \\/datum/seed/proc/mutate() called tick#: [world.time]")
+	var/mutation_type = pick_mut(severity)
+	apply_mut(mutation_type, severity)
+
+	return
+
+/obj/machinery/portable_atmospherics/hydroponics/proc/pick_mut(var/severity, var/mutation_category = "")
+
+	var/datum/seed/S = seed
+	if (!S) return
+
+	//First we'll pick a CATEGORY of mutations to look from, for simplicity and to keep an even ratio of good things to bad things if more mutations are added.
+	//This looks like shit, but it's a lot easier to read/change this way. Maybe. Promise. Hahaha. Shit.
+	if(!mutation_category) //If we already have a category, use that instead.
+		mutation_category = pick(\
+			15;"SomethingGood", \
+			min(0.4*severity, 7);"SomethingBad", \
+			min(8,  0.7*(severity-5));"SomethingWeird", \
+			min(7,  severity-12);"SomethingBizarre", \
+			min(14, severity-12);"SomethingAwful", \
+			min(20, severity-14);"SomethingDangerous"\
+			)
+	world << "We're going for [mutation_category] with severity [severity]"
+	var/mutation_type
+	//Now we'll pick a certain type of mutation from that category, special considerations in mind.
+	switch(mutation_category)
+		if("SomethingGood")
+			mutation_type = pick(\
+			10;	"plusstat_potency", \
+			S.yield == -1 ? 0 : 6;	"plusstat_yield",\
+			3;	"plusstat_weed&toxins_tolerance",\
+			5;	"plusstat_lifespan&endurance",\
+			5;	"plusstat_production&maturation",\
+			3;	"plusstat_heat&pressure_tolerance",\
+			3;	"plusstat_light_tolerance", \
+			3;	"plusstat_nutrient&water_consumption", \
+			S.yield != -1 && !S.harvest_repeat ? 0.4 : 0;	"toggle_repeatharvest",\
+			)
+		if("SomethingWeird")
+			mutation_type = pick(\
+			S.biolum ? 10 : 0;"biolum_changecolor",\
```

Comment:
It would be really nice if you could use spaces to align these things, so they're a bit more readable.